base/reflection.jl

```diff
@@ -428,6 +428,11 @@ function isprimitivetype(@nospecialize(t::Type))
     return !hasfield && t.size != 0 && !t.abstract
 end
 
+struct IncompleteTypeException
+    dt::DataType
+end
+must_be_complete(t::DataType) = (t.incomplete && throw(IncompleteTypeException(t)); t)
```

Comment:
doesn't seem like this should need a custom type (which should be subtype Exception if it was kept). Just `error("informative message here about ", dt.name.name)`